src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -2366,6 +2400,43 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
             }
 
+            case NI_Sve_LoadVectorFirstFaulting:
+            {
+                if (unspilledFfr)
+                {
+                    // We have unspilled the FFR in op1Reg. Restore it back in FFR register.
+                    GetEmitter()->emitIns_R(INS_sve_wrffr, emitSize, op1Reg, opt);
+                }
+
+                insScalableOpts sopt = (opt == INS_OPTS_SCALABLE_B) ? INS_SCALABLE_OPTS_NONE : INS_SCALABLE_OPTS_LSL_N;
+                GetEmitter()->emitIns_R_R_R_R(ins, emitSize, targetReg, op1Reg, op2Reg, REG_ZR, opt, sopt);
+                break;
+            }
+
+            case NI_Sve_GetFfrByte:
+            case NI_Sve_GetFfrInt16:
+            case NI_Sve_GetFfrInt32:
+            case NI_Sve_GetFfrInt64:
+            case NI_Sve_GetFfrSByte:
+            case NI_Sve_GetFfrUInt16:
+            case NI_Sve_GetFfrUInt32:
+            case NI_Sve_GetFfrUInt64:
+            {
+                if (unspilledFfr)
+                {
+                    // We have unspilled the FFR in op1Reg. Restore it back in FFR register.
+                    GetEmitter()->emitIns_R(INS_sve_wrffr, emitSize, op1Reg, opt);
```

Comment:
In this case it seems we could just emit `mov targetReg, op1Reg` and skip `rdffr` entirely?

Reply:
We could, but I will have to double check if that is fine to do, because sometimes, we do want to make sure `ffr` is up-to-date. I will leave this as a future optimization.